File: pkg/codegen/testing/test/testdata/aws-resource-options-4.26-pp/dotnet/aws-resource-options-4.26.cs
using System.Collections.Generic;
using Pulumi;
using Aws = Pulumi.Aws;

return await Deployment.RunAsync(() => 
{
    var provider = new Aws.Provider("provider", new()
    {
        Region = "us-west-2",
    });

    var bucket1 = new Aws.S3.Bucket("bucket1", new()
    {
    }, new CustomResourceOptions
    {
        Provider = provider,
        DependsOn = new[]
        {
            provider,
        },
        Protect = true,
        IgnoreChanges =
        {
            "bucket",
            "lifecycleRules[0]",
        }
    });

});

File: pkg/codegen/testing/utils/testdata/aws-resource-options-4.26-pp/nodejs/aws-resource-options-4.26.ts
import * as pulumi from "@pulumi/pulumi";
import * as aws from "@pulumi/aws";

const provider = new aws.Provider("provider", {region: "us-west-2"});
const bucket1 = new aws.s3.Bucket("bucket1", {}, {
    provider: provider,
    dependsOn: [provider],
    protect: true,
    ignoreChanges: [
        "bucket",
        "lifecycleRules[0]",
    ],
});
